program/src/processor.rs

```diff
@@ -150,3 +153,18 @@ pub fn process_instruction(
         }
     }
 }
+
+/// Given a DID, validate that the signers contain at least one
+/// account that has permissions to sign transactions using the DID.
+pub fn validate_owner(did: &AccountInfo, signers: &[AccountInfo]) -> ProgramResult {
+    let solid = program_borsh::try_from_slice_incomplete::<SolidData>(*did.data.borrow())?;
```

Comment:
You should do an additional check that the did `AccountInfo` is actually owned by the did program, otherwise someone can spoof a did!

Reply:
Eeek! - fixed